src/js/jsx/sections/pages/PagesPanel.jsx

```diff
@@ -379,35 +428,23 @@ define(function (require, exports, module) {
                 layerCount = null;
                 childComponents = null;
             } else {
-                layerComponents = doc.layers.top
+                layerComponents = doc.layers.allVisible.reverse()
                     .map(function (layer) {
-                        // If the layer or its descendants is not the dropTarget, pass false here
-                        // I am attempting to not have so many extra renders of the Layers...but this may be slower
-                        // Otherwise, pass down the layer
-
                         var dragTarget = this.props.dragTarget;
 
-                        var shouldPassDropTarget = !!(layer &&
-                                this.props.dropTarget &&
-                                doc.layers.descendants(layer).indexOf(this.props.dropTarget.keyObject) !== -1),
-                            shouldPassDragTarget = false;
-
-                        if (dragTarget) {
-                            // Should be false if the layer and the layers children do not appear in this list
-                            shouldPassDragTarget = dragTarget.some(function (dragT) {
-                                return layer && doc.layers.descendants(layer).indexOf(dragT) !== -1;
-                            });
-                        } else if (this.state.futureReorder) {
+                        if (this.state.futureReorder) {
                             dragTarget = this.props.pastDragTarget;
-                            shouldPassDragTarget = dragTarget.some(function (dragT) {
-                                return layer && doc.layers.descendants(layer).indexOf(dragT) !== -1;
-                            });
                         }
+                        var isDragTarget = !!(dragTarget && dragTarget.indexOf(layer) !== -1),
+                            isDropTarget = !!(this.props.dropTarget && this.props.dropTarget.keyObject === layer);
 
                         return (
                             <li key={layer.key}>
-                                <Layer
+                                <LayerFace
+                                    ref={layer.key}
                                     disabled={this.props.disabled}
+                                    registerOnMount={!this.state.batchRegister}
```

Comment:
In a perfect world, all the state elements in this component would be documented with a `getInitialState` function. Or at least the new ones...